config/config.go

```diff
@@ -84,6 +84,9 @@ type Configuration struct {
 	GenerateBidID bool `mapstructure:"generate_bid_id"`
 	// GenerateRequestID overrides the bidrequest.id in an AMP Request or an App Stored Request with a generated UUID if set to true. The default is false.
 	GenerateRequestID bool `mapstructure:"generate_request_id"`
+	// IntegrationTypesMap will hold the integration type values that the host will allow.
+	// Map structure isn't included here because the integration type feature is still being worked on, but once complete, this variable will be updated
+	IntegrationTypesMap map[string]struct{}
```

Comment:
Two questions. First is how do you plan on populating `IntegrationTypesMap`? For the above Blacklist maps, we have two entries, one a slice of strings, and the second is the actual map. In your configuration you would provide the list of values, and then config code builds the map from the list.

The second question is how do we default to integration types not being required? We don't want to break every install of PBS that is not currently requiring integration types to be sent in requests.

Reply:
To your first question, I like the idea of copying how the Blacklist maps work, so I could have one variable `IntegrationTypes` that is just a slice of strings indicating what IntegrationTypes the host company allows, and this slice can be filled with another `SetDefault` of that `SetupViper` function. And then that slice can be apart of the `IntegrationTypesMap`. And then we can continue to utilize the fast lookup the map allows us. 

To your second question, integration types are not being required in this PR since we never actually call any of the validate/get functions set up for integration types, so the flow of current PBS instances won't be broken. But in the future, we could create a simple boolean (i.e. `AllowIntegrationTypes`), that if false, avoids any evaluation of integration type information. Or host company's could choose to not populate the IntegrationTypes slice, and then we could add a check that makes it if IntegrationTypes slice is empty, we just continue evaluating the request ignoring IntegrationType related validation.

Reply:
Oh, yes, you don't actually call these new functions. Seems like a small enough addition, was there a reason that the actual activation was left to another PR?

Reply:
Since we still have to implement the integration type features for accounts (i.e. Default Integration), we can leave the activation for that PR!